### PR TITLE
doc [ci skip]

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: runtime/doc/if_lua.txt [hunk 1]
```diff
@@ -552,7 +568,17 @@ tsnode:named_descendant_for_range(start_row, start_col, end_row, end_col)
...
+
+vim.api.{func}({...})					*vim.api*
+        Invokes Nvim |API| function {func} with arguments {...}.
+        Example: call the "nvim_get_current_line()" API function: >
+            print(tostring(vim.api.nvim_get_current_line()))
+
+vim.call({func}, {...})					*vim.call()*
+        Invokes |vim-function| or |user-function| {func} with arguments {...}.
+        See also |vim.fn|. Equivalent to: >
+            vim.fn[func]({...})
```
Comment (justinmk) [inline runtime/doc/if_lua.txt]: we should try to keep this in alphabetical order. I didn't do that fully here, but just moved `vim.call` near the top.

**Comment by norcalli** [comment]
Does this just include a bunch of doc changes all in one? If so, I believe, `on_detach` for `nvim_buf_attach` is missing. Also on that function, I would add the note that returning true from the `on_lines` function will cause the event notifications for that handler to stop. It might be mentioned somewhere else, but it should *definitely* be included there anyway.

**Comment by mhinz** [comment]
https://github.com/neovim/neovim/commit/610755ff62049bd38cc72524a26ac0a27d951c3c#r35791892

**Comment by justinmk** [comment]
> `on_detach` for `nvim_buf_attach` is missing. Also on that function, I would add the note that returning true from the `on_lines` function will cause the event notifications for that handler to stop. It might be mentioned somewhere else, but it should _definitely_ be included there anyway.

done

**Pull request merged**